src/Models/Market.php

```diff
@@ -35,10 +36,15 @@ protected static function boot()
         parent::boot();
 
         static::creating(function (Market $market) {
-            $market->slug = $market->slug ?: Str::slug($market->city);
-            $invalidSlugs = config('locations.invalid_slugs') ?? [];
-            if (in_array($market->slug, $invalidSlugs)) {
-                $market->slug = Str::slug("{$market->slug}-{$market->state}");
+            $other_market = Market::where('slug', $market->slug)->first();
+
+            if ($other_market) {
+                throw new \Exception("there is a market with the slug selected");
+            }
+
+            if (empty($market->page_id)) {
+                $page = Page::query()->create(['slug' => $market->slug, 'title' => $market->title]);
+                $market->page_id = $page->id;
```

Comment:
@pdbreen thanks for all your feedbacks accross this PR, I've been fixing one by one.. however; I still having issues with the relations

after 

$page = Page::query()->create(['slug' => $market->slug, 'title' => $market->title]);
 $market->page_id = $page->id;

the object => `market` have the relation with `page` based on

    public function page()
    {
        return $this->belongsTo(Page::class);
    }

so, if I use $market->page->id // it should work , but it doesnt

notice If I `dd($market)`

![image](https://user-images.githubusercontent.com/13080491/113621035-1f6d1480-9629-11eb-8c24-a8fb21d9b255.png)

What I'm missing here?

Reply:
@wolfrednicolas  -- You have to fix the `Publishable` trait in the blog package.  Details

> Next - the Page model is using a Publishable trait that adds a global context requiring published_at to be less than now:
> 
>         static::addGlobalScope('published', static function (Builder $builder) {
>             $builder->where('published_at', '<', now());
>         });
> This should change to be less than or equal. Tests run fast enough that no time may have elapsed causing this condition to fail as written resulting in the page seeming to be missing. The isPublished() method in the trait should also be updated to be less than or equal as well.

Reply:
If you add `$this->logToStderr()->logSql();` - you can see the actual SQL being executed is automatically filtering by the `published_at` field because its registered as a global scope in the trait.